pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -604,7 +609,7 @@ func getDeviceMountPath(volume *reconstructedVolume) (string, error) {
 	return deviceMountPath, nil
 }
 
-func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*reconstructedVolume) error {
+func (rc *reconciler) updateStates(volumesNeedUpdate []*reconstructedVolume) error {
```

Comment:
This change should make it similar to any other plugin and I have not seen any issues with it so far with multiple vendor plugins.
Do we need the volumemanager changes?

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volumemanager/reconciler/reconciler.go#L414
`volumesNeedUpdate[reconstructedVolume.volumeName] = reconstructedVolume`

> GetVolumeName should return unique name in the cluster except for inline volumes.

The default Flexvolme `GetVolumeName` return `inline volume name`, and the pods on the node can use the same `inline volume name`.
When multiple volumes with same  `inline volume name` need to be cleaned up, only one can succeed if using `map volumesNeedUpdate`

`syncStates` is called once during the kubelet life cycle.

Reply:
@xiaoanyunfei This is a known limitation with inline & flex volumes. Inline volumes are not recommended for flex volumes. There is not enough information to reconstruct plugin information, unless we do Kubelet checkpointing.

Reply:
@chakri-nelluri 
The changes in `reconciler.go` don't affect other volume plugin, 
I just change the parameters format of `updateStates` and `updateDevicePath`.

 `This is a known limitation with inline & flex volumes. Inline volumes are not recommended for flex volumes.`

I don't get your point. Every flexvolume has an `inline volume name`, is that right?

So multiple flexvolumes in different pods can have the same `inline volume name`

If we have ten pods with flexVolumes which has same `inline volume name`(reconstructedVolume.volumeName in reconciler.syncStates)

 such as
>  name: hulklvm
     driver: hulk/lvm
      fsType: ext4
      options:
        size: 32Gi
        volumegroup: volumevg

when the pods become orphan pod, only one pod(in map `reconciler.syncStates volumesNeedUpdate`) can be clean up in the previous ` reconciler.go`

Reply:
@chakri-nelluri  

> Flexvolume inline volume names have to be cluster-wide unique if they are used. That is a known limitation

But k8s code don't limit this.  I think k8s should prepare for the worst.

Reply:
I will separate this file into another pr